### ISSUE TITLE
no-unused-properties reports exported objects when they are also used in the file

### ISSUE THREAD

**Issue by lukeapage** [sindresorhus/eslint-plugin-unicorn#215]
no-unused-properties has found many unused properties - great rule. However in this situation it is reporting errors..

```
export const ITEM_A = {
    id: 'a',
    fails: true,
};

export const ITEM_B = {
    id: ITEM_A.id,
};
```

here, `fails` is reported. If I don't use `ITEM_A.id`, then nothing is reported.
